.azure-pipelines/resolve_installer_location.py

```diff
@@ -0,0 +1,158 @@
+import platform
+import json
+import argparse
+import urllib
+import urllib.request
+from subprocess import check_call, CalledProcessError
+import sys
+import os
+import zipfile
+import tarfile
+import time
+
+from packaging.version import Version
+from packaging.version import parse
+from packaging.version import InvalidVersion
+
+import pdb
+
+# SOURCE OF THIS FILE: https://github.com/actions/python-versions
+# this is the official mapping file for gh-actions to retrieve python installers
+MANIFEST_LOCATION = "https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json"
+
+MAX_INSTALLER_RETRY = 3
+CURRENT_UBUNTU_VERSION = "18.04"  # full title is ubuntu-18.04
+
+UNIX_INSTALL_ARRAY = ["sh", "setup.sh"]
+WIN_INSTALL_ARRAY = ["pwsh", "setup.ps1"]
+
+
+def download_installer(remote_path, local_path):
+    retries = 0
+
+    while True:
+        try:
+            urllib.request.urlretrieve(remote_path, local_path)
+            break
+        except Exception as e:
+            print(e)
+            retries += 1
+
+            if retries >= MAX_INSTALLER_RETRY:
+                print(
+                    "Unable to recover after attempting to download {} {} times".format(
+                        remote_path, retries
+                    )
+                )
+                exit(1)
+            time.sleep(10)
+
+
+def install_selected_python_version(installer_url, installer_folder):
+    current_plat = platform.system().lower()
+
+    installer_folder = os.path.normpath(os.path.abspath(installer_folder))
+    if not os.path.exists(installer_folder):
+        os.mkdir(installer_folder)
```

Comment:
Do we want to handle any permissions issues on the installation directory? Or just let them raise?

Reply:
I don't know the answer, the scripts are provided by Scott.
but I think he should have already taken it into account :P cc @scbedd